[Upgrade Assistant] Update  UX for 7.17

sabarasaba opened the pull request:
Fixes https://github.com/elastic/kibana/issues/119738

### Summary
Given that we'll have another minor update before the next major, we need to tweak the UX of Upgrade Assistant to give the right message to our users. In order to achieve that, there is a few changes we'll have to do to the copy of the page to make it clearer that this is a preparation for the next major version:

* Change the copy in the header, step 3(Review deprecated settings) and step 4(get ready for upgrade)
* Change the copy in ES deprecation issue/logs and kibana deprecations
* Change the URL link for "What's new in 8.x" (which now points to what's new in 7.16)

Also, a few UI elements were hidden from the UI:
* Remove step 2(upgrade system indices) from the wizard
* Remove both CTA (Cloud and on prem) in Step 4(get ready for upgrade)

### About system indices migration

Initially the Upgrade Assistant in 7.16 was designed specifically for upgrading to the next major. But with the addition of the 7.17 release, we decided to only leave the necessary tools to the user to assist him with the upcoming minor upgrade. We then ended up removing the system indices migration from the UI since it was designed to assist users into major upgrades and not minors in an effort to align it more with some of the thoughts explained with https://github.com/elastic/kibana/issues/117220.

With these changes **we expect** users to upgrade to 7.17 before actually migrating to the next major, as we always guide them to do. Note that I removed the entry point for system indices management from the code and skipped the tests, the code wont get into the bundle since its not required anywhere.

**Screenshots**

<details>
<summary>Overview of changes</summary>


![Screenshot 2021-11-30 at 09 14 58](https://user-images.githubusercontent.com/1191206/144010081-6f303cac-729a-45f8-8451-9de25e4b511f.png)
![Screenshot 2021-11-30 at 09 14 53](https://user-images.githubusercontent.com/1191206/144010088-5b878b2b-7f6e-4992-96f1-1a86a0d29210.png)
![Screenshot 2021-11-30 at 09 14 47](https://user-images.githubusercontent.com/1191206/144010093-4bdc6ecb-e493-44b8-b141-f5d1f3f59185.png)
![Screenshot 2021-11-30 at 09 14 40](https://user-images.githubusercontent.com/1191206/144010098-6a8444cb-8020-41f2-82ef-9e55abbe41a0.png)
</details>

elasticmachine commented:
Pinging @elastic/kibana-stack-management (Team:Stack Management)

sebelga commented on x-pack/plugins/upgrade_assistant/public/application/components/kibana_deprecations/kibana_deprecations.tsx:
```diff
@@ -30,7 +30,8 @@ const i18nTexts = {
   pageDescription: i18n.translate('xpack.upgradeAssistant.kibanaDeprecations.pageDescription', {
-    defaultMessage: 'Resolve all critical issues before upgrading.',
+    defaultMessage:
+      'Resolve all critical issues before upgrading to the next major version of Elastic.',
```
Again, but let's see what @debadair thinks, I'd put "to prepare the upgrade to"

jrodewig replied:
I don't think "to prepare" is necessary. I think it's understood that you're preparing to upgrade.

sebelga commented on x-pack/plugins/upgrade_assistant/public/application/components/overview/fix_issues_step/fix_issues_step.tsx:
```diff
@@ -105,7 +105,7 @@ export const getFixIssuesStep = ({
             <FormattedMessage
               id="xpack.upgradeAssistant.overview.fixIssuesStepDescription"
-              defaultMessage="You must resolve any critical Elasticsearch and Kibana configuration issues before upgrading to Elastic 8.x. Ignoring warnings might result in differences in behavior after you upgrade. {accessDeprecationLogsMessage}"
+              defaultMessage="You can start resolving any critical Elasticsearch and Kibana configuration issues before upgrading to the next major version of Elastic, ignoring warnings might result in differences in behavior. {accessDeprecationLogsMessage}"
```
"...configuration issues to prepare the upgrade to ..."

jrodewig replied:
There are some grammatical issues with the current copy:

```suggestion
              defaultMessage="You can start resolving any critical Elasticsearch and Kibana configuration issues before upgrading to the next major version of Elastic. Ignoring warnings might result in differences in behavior. {accessDeprecationLogsMessage}"
```

I'd make the copy a bit more imperative:

```suggestion
              defaultMessage="Resolve any critical Elasticsearch and Kibana configuration issues before upgrading to the next major version. Ignoring warnings might result in unexpected behavior. {accessDeprecationLogsMessage}"
```

sebelga commented on x-pack/plugins/upgrade_assistant/public/application/components/overview/upgrade_step/upgrade_step.tsx:
```diff
@@ -7,119 +7,33 @@
   upgradeStepDescription: i18n.translate('xpack.upgradeAssistant.overview.upgradeStepDescription', {
     defaultMessage:
-      'Once you’ve resolved all critical issues and verified that your applications are ready, you can upgrade to Elastic 8.x. Be sure to back up your data again before upgrading.',
+      'You can start resolving all critical issues in preparation for the upcoming Elastic 8.x. Be sure to back up your data again before upgrading',
```
Here you used "in preparation" 😊 It could also be "...to prepare for".

jrodewig replied:
The current copy is missing a period:
```suggestion
      'You can start resolving all critical issues in preparation for the upcoming Elastic 8.x. Be sure to back up your data again before upgrading.',
```

This seemed a little odd to me as it came _after_ step 2. At this point, the user should already have solved their critical issues, right? 

```suggestion
      'If you've resolved all critical issues and checked your deprecation logs, you're ready for the next major version of Elastic. Check this page again when you upgrade to Elastic 8.x.',
```

<img width="1230" alt="Screen Shot 2021-11-29 at 10 38 51 AM" src="https://user-images.githubusercontent.com/40268737/143897548-7e6fecbe-4743-4675-b790-b70ea16613ff.PNG">

cjcenizal replied:
++ to James's second suggestion. "Be sure to back up your data again before upgrading" seems out of place, given that we don't expect the user to upgrade to 8 from 7.16. "Check this page again..." guides the user to the correct next step in the process.

sebelga reviewed (approved):
LGTM! I tested locally and works as expected. 
I left a few note around using "to _prepare for_ the upgrade" (which is the terminology [our docs](https://www.elastic.co/guide/en/kibana/7.16/upgrade-assistant.html) use).

sabarasaba commented:
Thanks for having a look @sebelga! I've addressed your feedback about the tests and docs link. I'll defer on the copy changes to the docs team 🚀

sabarasaba commented:
@elasticmachine merge upstream

jrodewig commented on x-pack/plugins/upgrade_assistant/public/application/components/es_deprecation_logs/fix_deprecation_logs/fix_deprecation_logs.tsx:
```diff
@@ -38,7 +38,7 @@ const i18nTexts = {
     <FormattedMessage
       id="xpack.upgradeAssistant.overview.apiCompatibilityNoteBody"
-      defaultMessage="We recommend you resolve all deprecation issues before upgrading. If needed, you can apply API compatibility headers to requests that use deprecated features. {learnMoreLink}."
+      defaultMessage="You can start resolving deprecation issues before upgrading to the next major version of Elastic. If needed, you can apply API compatibility headers to requests that use deprecated features. {learnMoreLink}."
```
I think we should leave this largely as is and just append "to the next major version of Elastic." We're trying to present the API compatibility headers as a less preferred safety mechanism. It doesn't really work without "We recommend..."

```suggestion
      defaultMessage="We recommend you resolve all deprecation issues before upgrading to the next major version of Elastic. If needed, you can apply API compatibility headers to requests that use deprecated features. {learnMoreLink}."
```
<img width="1211" alt="Screen Shot 2021-11-29 at 10 12 17 AM" src="https://user-images.githubusercontent.com/40268737/143893067-f3f8d796-4157-4e35-acc9-2b3d8fa8728e.PNG">

jrodewig commented on x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/es_deprecations.tsx:
```diff
@@ -46,7 +46,7 @@ const i18nTexts = {
   pageDescription: i18n.translate('xpack.upgradeAssistant.esDeprecations.pageDescription', {
     defaultMessage:
-      'Resolve all critical issues before upgrading. Before making changes, ensure you have a current snapshot of your cluster. Indices created before 7.0 must be reindexed or removed.',
+      'You can start resolving all critical issues before upgrading to the next major version of Elastic. Before making changes, ensure you have a current snapshot of your cluster. Indices created before 7.0 will have to be reindexed or removed.',
```
I think stating "to the next major version of Elastic" adds enough clarity here. While it's potentially a lot of work, there's no real harm to resolving critical issues or reindexing 7.0 indices before 7.last.

```suggestion
      'Resolve all critical issues before upgrading to the next major version of Elastic. Before making changes, make sure you have a current snapshot of your cluster. To upgrade to 8.x, you must reindex or remove indices created before 7.0.',
```

jrodewig commented on x-pack/plugins/upgrade_assistant/public/application/components/overview/upgrade_step/upgrade_step.tsx:
```diff
@@ -7,119 +7,33 @@
 const i18nTexts = {
   upgradeStepTitle: i18n.translate('xpack.upgradeAssistant.overview.upgradeStepTitle', {
-    defaultMessage: 'Upgrade to Elastic 8.x',
+    defaultMessage: 'Get ready for the next version of Elastic',
```
This steps seems odd to me. Isn't the entire page + other steps about getting ready for the upgrade?

I'd advise the user to check back at this page when it's time to upgrade to 8.x.

```suggestion
    defaultMessage: 'Check back here when it's time to upgrade to 8.0',
```

jrodewig replied:
@sabarasaba What do you think about implementing this new step title or something similar?

jrodewig reviewed (commented):
Thanks for making these changes @sabarasaba. I noted some grammatical issues with a few bits of the copy. I also don't think we should go overboard with "to prepare," etc. here as long as it's clear that the steps are related to a major version upgrade.

My only large piece of feedback is related to step 3. I think we should update this step to advise users to come back to the page when they're ready to upgrade to 8.x. The current wording is kind of circular.

sebelga commented:
@sabarasaba I think this is going to `7.16.0` (for the label you put). @cjcenizal can you confirm?

cjcenizal commented:
@sebelga Correct, AFAIK we want this to ship with 7.16.0.

cjcenizal reviewed (approved):
💥 These changes LGTM. My only concerns are addressed by James's suggestions. Thanks for the fast turnaround on this!  We will cut a new BC on Thursday, so let's aim to merge this by Wednesday, 5PM Pacific.

sabarasaba commented:
Thanks for having a look @jrodewig! I've addressed your feedback with https://github.com/elastic/kibana/pull/119798/commits/1b221d3b0c24fee02916a467327ee7bea6c3d418 and also updated the screenshots from the PR description.

jrodewig reviewed (approved):
I didn't test locally, but this LGTM based on the screenshots and code. There was one suggestion that may have been overlooked, but it's not blocking either way: https://github.com/elastic/kibana/pull/119798#discussion_r758478598

Thanks for the fast work on this @sabarasaba! 🚀
